CMakeLists.txt

```diff
@@ -363,6 +358,48 @@ add_swift_library(FoundationNetworking
                     $<$<PLATFORM_ID:Windows>:CoreFoundationResources>
                     Foundation)
 
+add_swift_library(FoundationXML
+                  MODULE_NAME
+                    FoundationXML
+                  MODULE_LINK_NAME
+                    FoundationXML
+                  MODULE_PATH
+                    ${CMAKE_CURRENT_BINARY_DIR}/swift/FoundationXML.swiftmodule
+                  SOURCES
+                  Foundation/XMLDocument.swift
+                  Foundation/XMLDTD.swift
+                  Foundation/XMLDTDNode.swift
+                  Foundation/XMLElement.swift
+                  Foundation/XMLNode.swift
+                  Foundation/XMLParser.swift
+                  TARGET
+                    ${CMAKE_C_COMPILER_TARGET}
+                  LINK_FLAGS
+                    ${MSVCRT_LINK_FLAGS}
+                    -L${CMAKE_CURRENT_BINARY_DIR}
+                    ${libdispatch_ldflags}
+                    -lFoundation
+                    ${Foundation_INTERFACE_LIBRARIES}
+                    ${CFXMLInterface_LIBRARIES}
+                    ${LIBXML2_LIBRARIES}
+                    ${Foundation_RPATH}
+                    ${WORKAROUND_SR9138}
+                    ${WORKAROUND_SR9995}
+                  SWIFT_FLAGS
+                    -DDEPLOYMENT_RUNTIME_SWIFT
+                    -DNS_BUILDING_FOUNDATION_NETWORKING
```

Comment:
Is this one correct? Or necessary? I don't see it used anywhere else in the commit (unless it appears in some unexpanded file). If we rely on it because it was already used like this in FoundationNetworking, maybe it would be good to update it to something more “neutral”. If anyone knows the answer, I will not mind doing the PR, but I don’t exactly know what to do.

Reply:
That was, indeed, a mistake.

There is one file that is shared between the Foundation and FoundationNetworking builds containing internal calls; in Networking, we have to import Foundation, and that's what this guards. It is unnecessary in FoundationXML.